EssentialCSharp.Web/Areas/Identity/Pages/Account/Manage/ChangePassword.cshtml.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#nullable disable
+
+using System.ComponentModel.DataAnnotations;
+using EssentialCSharp.Web.Areas.Identity.Data;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace EssentialCSharp.Web.Areas.Identity.Pages.Account.Manage
+{
+    public class ChangePasswordModel : PageModel
+    {
+
+        private readonly UserManager<EssentialCSharpWebUser> _userManager;
+        private readonly SignInManager<EssentialCSharpWebUser> _signInManager;
+        private readonly ILogger<ChangePasswordModel> _logger;
+
+        public ChangePasswordModel(
+            UserManager<EssentialCSharpWebUser> userManager,
+            SignInManager<EssentialCSharpWebUser> signInManager,
+            ILogger<ChangePasswordModel> logger)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _logger = logger;
+        }
+
+        [BindProperty]
+        public InputModel Input { get; set; }
+
+        [TempData]
+        public string StatusMessage { get; set; }
+
+        public class InputModel
```

Comment:
We should probably consider not using a public nested class. In general public classes should very rarely be nested.

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1034

Reply:
I agree, except I don't expect anyone else to instantiate this class except myself if I am pre-populating data or whatever binds the data to it from the front-end, so I think it is fine in this case? I think if I made it private, it wouldn't be able to bind properly from the frontend.

Thoughts?